pymchelper/readers/topas.py

```diff
@@ -0,0 +1,172 @@
+import os
+import re
+
+import numpy as np
+from pymchelper.axis import MeshAxis
+from pymchelper.page import Page
+from pymchelper.readers.common import Reader, ReaderFactory
+
+
+class TopasReaderFactory(ReaderFactory):
+    """Factory for TopasReader"""
+
+    def get_reader(self):
+        """Return TopasReader if the file extension is .csv"""
+        if self.filename.endswith('.csv'):
+            return TopasReader
+        return None
+
+
+class TopasReader(Reader):
+    """Reader for Topas output files"""
+
+    def __init__(self, filename):
+        super(TopasReader, self).__init__(filename)
+        self.directory = os.path.dirname(filename)
+
+    def get_bins(self, dimensions, bins_data, output_data):  # skipcq: PYL-R0201
+        """Get number of bins, bin size and unit for each dimension"""
+        for dimension in dimensions:
+            pattern = f"# {dimension} in (\\d+) bin[s ] of ([\\d.]+) (\\w+)"
+            match = re.search(pattern, output_data)
+            if match:
+                bins_data[dimension] = {'num': int(match.group(1)),
+                                        'size': float(match.group(2)),
+                                        'unit': match.group(3)}
+            else:
+                return False
+        return True
+
+    def get_scorer_name(self, output_data):  # skipcq: PYL-R0201
```

Comment:
```suggestion
    @staticmethod
    def get_scorer_name(output_data):
```

add type hinting as well